app/controllers/stream_entries_controller.rb

```diff
@@ -19,6 +20,8 @@ def show
       end
 
       format.atom do
+        skip_session! unless @stream_entry.hidden?
+        expires_in 3.minutes, public: true
```

Comment:
`public: !@stream_entry.hidden?, private: @stream_entry.hidden?`

If you do not do as above, does private toot caching on CDN?

Reply:
In practice, this shouldn't happen (because the session cookie is still set, preventing caching, and because a logged-in user requesting a private toot this way is extremely unlikely), but it is something I have overlooked, you're right.